### PR TITLE
Enhance test-content command

### PR DESCRIPTION
**Pull request by eyalpalo**
## Related Issues
relates: https://jira-dc.paloaltonetworks.com/browse/CIAC-10132

## Description
Adjusted the `test-content` command to work properly with multiple cloud servers.

[SDK Nightly on master](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/1169482)
[SDK Nightly on Infra changes](https://gitlab.xdr.pan.local/xdr/cortex-content/content/-/pipelines/1168577)

### CODE REVIEW

File: .changelog/4321.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Modified the **test-content** command which would previously behave incorrectly under certain conditions.
```
Comment (michal-dagan) [inline .changelog/4321.yml:2]:
```suggestion
- description: Added support to the (internal) **test-content** command to work properly with multiple cloud servers.
```

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 2]
```diff
@@ -2467,713 +2915,429 @@ def _get_failed_stage(
...
 
-    def _update_playbook_status(
-        self,
-        status: str,
-        is_first_playback_run: bool = False,
-        is_second_playback_run: bool = False,
-        is_record_run: bool = False,
-    ) -> str:
+    @staticmethod
+    def _get_circle_memory_data() -> Tuple[str, str]:
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:2954]: i think we can drop the circle memory parts from the code

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 3]
```diff
@@ -2467,713 +2915,429 @@ def _get_failed_stage(
...
-                status = self._update_playbook_status(status, is_record_run=True)
-                result_holder[RESULT] = status == PB_Status.SECOND_PLAYBACK_REQUIRED
-        # Running playback after successful record to verify the record is valid for future runs
-        if status == PB_Status.SECOND_PLAYBACK_REQUIRED:
-            self.playbook.log_info(
-                f"------ Test {self} start ------ (Mock: Second playback)"
+            stdout, stderr = self._get_circle_processes_data()
+            text = stderr or stdout
+            self._send_slack_message(
+                SLACK_MEM_CHANNEL_ID, text, "Content CircleCI", "False"
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:2996]: same here, drop the slack/circle code

**Review by kobymeir — APPROVED** [review]

File: demisto_sdk/commands/test_content/TestContentClasses.py [hunk 4]
```diff
@@ -817,239 +760,97 @@ def print_context_to_log(self, client: DefaultApi, incident_id: str):
...
-            test_configuration: A test configuration from content conf.json file.
-
-        Returns:
-            A list with integration names
+        Create servers object based on build type.
         """
-        tested_integrations = test_configuration.get("integrations", [])
-        if isinstance(tested_integrations, list):
-            return tested_integrations
+        servers: set = set()
```
Comment (kobymeir) [inline demisto_sdk/commands/test_content/TestContentClasses.py:832]: classic to change to set comprehension

**Review by kobymeir — APPROVED** [review]

**Review by yucohen — APPROVED** [review]

**Review by kobymeir — COMMENTED** [review]
approved

**Pull request merged**